packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -377,10 +382,37 @@ export class TestObjectProvider implements ITestObjectProvider {
 		requestHeader?: IRequestHeader,
 	): Promise<IContainer> {
 		const loader = this.createLoader([[defaultCodeDetails, entryPoint]], loaderProps);
-		return loader.resolve({
+
+		// Once ADO#3889 is done to switch default connection mode to "read" on load, we don't need
+		// to load "delayed" across the board. Remove the following code.
+		const delayConnection =
+			requestHeader === undefined || requestHeader[LoaderHeader.reconnect] !== false;
+		const headers: IRequestHeader = delayConnection
+			? {
+					[LoaderHeader.loadMode]: { deltaConnection: "delayed" },
+					...requestHeader,
+			  }
+			: requestHeader;
+
+		const container = await loader.resolve({
 			url: await this.driver.createContainerUrl(this.documentId),
-			headers: requestHeader,
+			headers,
 		});
+
+		// Once ADO#3889 is done to switch default connection mode to "read" on load, we don't need
+		// to load "delayed" across the board. Remove the following code.
+
+		if (delayConnection) {
+			// Older version might have the connect function, but that also means that delayed doesn't work
```

Comment:
can we make the current version a bit more type safe: `const maybeContainer = container as Partial<IContainer>` its just nice to get compile breaks it something changes in the future around this api.

Reply:
Done.